---
layout: project
title: "Proteogenomic Analysis"
tagline: ""
handle: proteogenomic-analysis
image: 
category: project
tags: []
---
{% include JB/setup %}

Over the past decades, many post-translational modifications (PTMs), such as ubiquitination, phosphorylation, and acetylation, have been studied for their role in regulating cell signaling events that are key for all cellular physiological functions. Recent advances in mass spectrometry (MS) technologies have enabled measurement of protein levels as well as protein modifications across cancer. Most studies thus far have focused on a single type of modification in a specific tumor type. We seek to understand the joint underlying patterns of PTMs in molecular signaling pathways that are shared across multiple cancer types by studying changes in protein acetylation and phosphorylation. We are part of the NCIâ€™s Clinical Proteomic Tumor Analysis Consortium ([CPTAC]) efforts in collaboration with Drs. D.R. Mani, Steven Carr, Lewis Cantley, and Li Ding, among others.

We previously developed innovative computational methods called CLUMPS (clustering of mutations in protein structures) that identifies significant spatial clustering of mutations within the protein 3D structure and surface ([Kamburov, et al., PNAS 2015]). The key advantage of CLUMPS is that it takes into account the 3D distance between amino acids and not just the 1D distance along the linear genome, and the fact that it uses homology modeling to map proteins to 3D structures. We are now applying a similar approach to identify clusters of altered PTM sites.

[CPTAC]: https://proteomics.cancer.gov/programs/cptac
[Kamburov, et al., PNAS 2015]: /papers/paper/clumps
