---
layout: project
title: "Driver discovery in non-coding regions of the genome"
tagline:
handle: non_coding_drivers
image: 
category: project
tags:
---
{% include JB/setup %}

Beyond being a major contributor to The Cancer Genome Atlas (TCGA) projects, leading various analyses in many of the TCGA manuscripts (including co-chairing the papillary thyroid cancer paper, the first to use whole-genomes to study tumors without coding drivers ([Integrated genomic characterization of papillary thyroid cancer, Cell 2014])), we helped lead TCGA into its next phase with the Pan-cancer Analysis of Whole Genomes (PCAWG) effort of TCGA and ICGC, contributing to several of the 23 papers released across the Nature journals in February 2020, including the flagship paper ([The ICGC/TCGA Pan-Cancer Analysis of Whole Genomes Consortium, Nature 2020]). In one of our studies, we searched for drivers in the non-coding regions (~99% of the genome) in whole genomes from cancer patients by employing our statistically rigorous strategies to analyze both point mutations and structural variants ([Rheinbay, Nielsen, Abascal, Wala, Shapira, et al. Nature 2020]). Although we discovered a few novel non-coding driver genes (e.g., point mutations in the 5â€² region of TP53), our study revealed that cancer drivers in non-coding regions are relatively rare, with the vast majority of drivers (87%) occurring in just the ~1% of the genome that is protein-coding.

[Integrated genomic characterization of papillary thyroid cancer, Cell 2014]: /papers/paper/papillary-thyroid-carcinoma
[The ICGC/TCGA Pan-Cancer Analysis of Whole Genomes Consortium, Nature 2020]: /papers/paper/pcawg-flagship
[Rheinbay, Nielsen, Abascal, Wala, Shapira, et al. Nature 2020]: /papers/paper/non-coding-drivers
