---
layout: paper
title: "Why highly expressed proteins evolve slowly"
year: "2005"
shortref: "Drummond <i>et al. PNAS</i> 2005b"
nickname: why-highly-expressed
journal: "Proc. Nat'l. Acad. Sci. USA"
volume: 102
issue: 40
pages: 14338--43
authors: "Drummond DA, Bloom JD, Adami C, Wilke CO, and Arnold FH"
image: /assets/images/papers/default-paper.svg
pdf: 
pdflink: "http://www.pnas.org/content/102/40/14338.full.pdf?with-ds=yes"
github: 
pmid: 16176987
pmcid: PMC1242296
f1000: http://f1000.com/prime/1029543
figshare: 
doi: "10.1073/pnas.0504070102"
category: paper
peerreview: true
review: false
published: true
tags: [translation, yeast, translational robustness, gene expression, evolutionary rate, codon usage]
---
{% include JB/setup %}

# Abstract 

Much recent work has explored molecular and population-genetic constraints on the rate of protein sequence evolution. The best predictor of evolutionary rate is expression level, for reasons that have remained unexplained. Here, we hypothesize that selection to reduce the burden of protein misfolding will favor protein sequences with increased robustness to translational missense errors. Pressure for translational robustness increases with expression level and constrains sequence evolution. Using several sequenced yeast genomes, global expression and protein abundance data, and sets of paralogs traceable to an ancient whole-genome duplication in yeast, we rule out several confounding effects and show that expression level explains roughly half the variation in Saccharomyces cerevisiae protein evolutionary rates. We examine causes for expression's dominant role and find that genome-wide tests favor the translational robustness explanation over existing hypotheses that invoke constraints on function or translational efficiency. Our results suggest that proteins evolve at rates largely unrelated to their functions and can explain why highly expressed proteins evolve slowly across the tree of life.

